lib/fluent/plugin/out_google_cloud.rb

```diff
@@ -1238,6 +1247,15 @@ def determine_group_level_monitored_resource_and_labels(tag,
           # TODO(qingling128): Fix this temporary renaming from 'gke_container'
           # to 'container'.
           resource.type = 'container' if resource.type == 'gke_container'
+        else
+          # TODO(qingling128): This entire else clause is temporary before we
+          # implement buffering and caching.
+          @log.error('Failing to retrieve monitored resource from Metadata' \
+                     " Agent with local_resource_id #{local_resource_id}. Try" \
+                     ' to construct resource locally if it is a k8s resource.')
```

Comment:
I think it makes sense to document the reason for constructing that fallback resource along with the notification, rather than separately inform the user "there's a failure, trying to construct a resource if we are in the right context" and then "we've constructed that resource" (or silence, if we aren't in the right context). So I would still suggest first calling `construct_k8s_resource_locally`, and then printing the message with both the reason and the resource.

Reply:
Changed. PTAL.